gateway/src/services/config-manager-v2.ts

```diff
@@ -267,10 +323,31 @@ export class ConfigManagerV2 {
   loadConfigRoot(configRootPath: string) {
     // Load the config root file.
     const configRootFullPath: string = fs.realpathSync(configRootPath);
+    const configRootTemplateFullPath: string = path.join(
+      ConfigTemplatesDir,
+      'root.yml'
+    );
     const configRootDir: string = path.dirname(configRootFullPath);
     const configRoot: ConfigurationRoot = yaml.load(
       fs.readFileSync(configRootFullPath, 'utf8')
     ) as ConfigurationRoot;
+    const configRootTemplate: ConfigurationRoot = yaml.load(
+      fs.readFileSync(configRootTemplateFullPath, 'utf8')
+    ) as ConfigurationRoot;
+
+    // version control to only handle upgrades
+    if (configRootTemplate.version > configRoot.version) {
+      // run migration in order if available
+      for (
```

Comment:
We may have to think about this again. saving after call to `addNamespace()` wouldn't work because configurations are loaded during call to `addNamespace()`.

Reply:
The way I imagined it is that saving/updating will be done during migration. I was thinking that was the major reason you wanted functions like `initiateWithTemplate` exported for use in the migration file.